gunicorn/arbiter.py

```diff
@@ -460,20 +460,27 @@ def reap_workers(self):
                     if exitcode == self.APP_LOAD_ERROR:
                         reason = "App failed to load."
                         raise HaltServer(reason, self.APP_LOAD_ERROR)
-                    worker = self.WORKERS.pop(wpid, None)
-                    if not worker:
-                        continue
-                    worker.tmp.close()
+                    self.reap_dead_worker(wpid)
```

Comment:
Note that this did not call `self.cfg.worker_exit(self, worker)` before, but will now. I believe that was an oversight but could use a double-check if it was intentional.